dask/Dockerfile

```diff
@@ -0,0 +1,25 @@
+FROM coffeateam/coffea-base:latest
+
+# Dask + Distributed
+RUN conda install --yes --freeze-installed \
+      -c conda-forge \
+      conda-build \
+      python-blosc \
+      cytoolz \
+      bokeh==2.2.3 \
+      dask==2.30.0 \
+      distributed==2.30.0 \
+      lz4 \
+      nomkl \
+      numpy==1.19.2 \
```

Comment:
I think we should leave all other dependencies as they were from base image, i.e. just add dask and distributed.

Reply:
+ any additional (dask-dataframe?)

Reply:
So no these ones: https://github.com/dask/dask-docker/blob/master/base/Dockerfile#L9-L12

Reply:
Yeah, a fair number of those (numpy, mpl, scipy, etc.) should be brought in as coffea dependencies, pinning them here may force an unnecessary version change, which makes the image bigger

Reply:
I will rechecked, before was like this
```
Downloading and Extracting Packages
async-timeout-3.0.1  | 11 KB     | ########## | 100% 
simpervisor-0.3      | 8 KB      | ########## | 100% 
cytoolz-0.11.0       | 396 KB    | ########## | 100% 
blosc-1.20.1         | 874 KB    | ########## | 100% 
toolz-0.11.1         | 46 KB     | ########## | 100% 
python-blosc-1.9.2   | 382 KB    | ########## | 100% 
typing_extensions-3. | 25 KB     | ########## | 100% 
dask-core-2.30.0     | 639 KB    | ########## | 100% 
zict-2.0.0           | 10 KB     | ########## | 100% 
nomkl-1.0            | 4 KB      | ########## | 100% 
typing-extensions-3. | 8 KB      | ########## | 100% 
partd-1.1.0          | 17 KB     | ########## | 100% 
click-7.1.2          | 64 KB     | ########## | 100% 
numpy-1.19.2         | 5.3 MB    | ########## | 100% 
yarl-1.6.3           | 142 KB    | ########## | 100% 
dask-2.30.0          | 4 KB      | ########## | 100% 
sortedcontainers-2.3 | 26 KB     | ########## | 100% 
matplotlib-3.3.2     | 6 KB      | ########## | 100% 
aiohttp-3.7.3        | 619 KB    | ########## | 100% 
locket-0.2.0         | 6 KB      | ########## | 100% 
msgpack-python-1.0.0 | 93 KB     | ########## | 100% 
bokeh-2.2.3          | 7.0 MB    | ########## | 100% 
tini-0.18.0          | 12 KB     | ########## | 100% 
tblib-1.6.0          | 14 KB     | ########## | 100% 
distributed-2.30.0   | 1.1 MB    | ########## | 100% 
multidict-4.7.5      | 71 KB     | ########## | 100% 
jupyter-server-proxy | 17 KB     | ########## | 100% 
heapdict-1.0.1       | 7 KB      | ########## | 100% 
matplotlib-base-3.3. | 6.7 MB    | ########## | 100% 
fsspec-0.8.4         | 65 KB     | ########## | 100% 
```
 now: 
```
blosc-1.20.1         | 874 KB    | ########## | 100% 
yarl-1.6.3           | 142 KB    | ########## | 100% 
zict-2.0.0           | 10 KB     | ########## | 100% 
partd-1.1.0          | 17 KB     | ########## | 100% 
toolz-0.11.1         | 46 KB     | ########## | 100% 
bokeh-2.2.3          | 7.0 MB    | ########## | 100% 
dask-2.30.0          | 4 KB      | ########## | 100% 
distributed-2.30.0   | 1.1 MB    | ########## | 100% 
tblib-1.6.0          | 14 KB     | ########## | 100% 
click-7.1.2          | 64 KB     | ########## | 100% 
fsspec-0.8.4         | 65 KB     | ########## | 100% 
typing_extensions-3. | 25 KB     | ########## | 100% 
simpervisor-0.3      | 8 KB      | ########## | 100% 
sortedcontainers-2.3 | 26 KB     | ########## | 100% 
async-timeout-3.0.1  | 11 KB     | ########## | 100% 
aiohttp-3.7.3        | 619 KB    | ########## | 100% 
multidict-4.7.5      | 71 KB     | ########## | 100% 
python-blosc-1.9.2   | 382 KB    | ########## | 100% 
cytoolz-0.11.0       | 396 KB    | ########## | 100% 
typing-extensions-3. | 8 KB      | ########## | 100% 
tini-0.18.0          | 12 KB     | ########## | 100% 
msgpack-python-1.0.0 | 93 KB     | ########## | 100% 
heapdict-1.0.1       | 7 KB      | ########## | 100% 
dask-core-2.30.0     | 639 KB    | ########## | 100% 
jupyter-server-proxy | 17 KB     | ########## | 100% 
locket-0.2.0         | 6 KB      | ########## | 100% 
nomkl-1.0            | 4 KB      | ########## | 100%  ```

Reply:
Just pip install dask[dataframe], since coffea supports dataframe-based ops now it should be in the worker image.

Reply:
@lgray I would prefer to use conda as much as possible, sinse we can updating image according to the latest version of Dask in conda.

Reply:
@lgray I just checked and looks like conda install all dependencies from `dask[dataframe]`: https://github.com/dask/dask/blob/master/setup.py#L18-L23

Reply:
adding pip is just doing nothing: ``` Step 3/4 : RUN /opt/conda/bin/pip install --no-cache-dir     dask[dataframe]==2.30.0
 ---> Running in 50bd9869a441
Requirement already satisfied: dask[dataframe]==2.30.0 in /opt/conda/lib/python3.8/site-packages (2.30.0)
Requirement already satisfied: pyyaml in /opt/conda/lib/python3.8/site-packages (from dask[dataframe]==2.30.0) (5.3.1)
Requirement already satisfied: partd>=0.3.10; extra == "dataframe" in /opt/conda/lib/python3.8/site-packages (from dask[dataframe]==2.30.0) (1.1.0)
Requirement already satisfied: fsspec>=0.6.0; extra == "dataframe" in /opt/conda/lib/python3.8/site-packages (from dask[dataframe]==2.30.0) (0.8.4)
Requirement already satisfied: numpy>=1.13.0; extra == "dataframe" in /opt/conda/lib/python3.8/site-packages (from dask[dataframe]==2.30.0) (1.19.4)
Requirement already satisfied: toolz>=0.8.2; extra == "dataframe" in /opt/conda/lib/python3.8/site-packages (from dask[dataframe]==2.30.0) (0.11.1)
Requirement already satisfied: pandas>=0.23.0; extra == "dataframe" in /opt/conda/lib/python3.8/site-packages (from dask[dataframe]==2.30.0) (1.1.4)
Requirement already satisfied: locket in /opt/conda/lib/python3.8/site-packages (from partd>=0.3.10; extra == "dataframe"->dask[dataframe]==2.30.0) (0.2.0)
Requirement already satisfied: python-dateutil>=2.7.3 in /opt/conda/lib/python3.8/site-packages (from pandas>=0.23.0; extra == "dataframe"->dask[dataframe]==2.30.0) (2.8.1)
Requirement already satisfied: pytz>=2017.2 in /opt/conda/lib/python3.8/site-packages (from pandas>=0.23.0; extra == "dataframe"->dask[dataframe]==2.30.0) (2020.4)
Requirement already satisfied: six>=1.5 in /opt/conda/lib/python3.8/site-packages (from python-dateutil>=2.7.3->pandas>=0.23.0; extra == "dataframe"->dask[dataframe]==2.30.0) (1.15.0)
Removing intermediate container 50bd9869a441
 ```

Reply:
ok - great! :-)